specs/ExtendedProcessFailed.md

```diff
@@ -0,0 +1,686 @@
+            foreach (CoreWebView2FrameInfo frameInfo in e.ImpactedFramesInfo)
+            {
+                // Sample virtual host name for the app's content.
+                string virtualAppHostName = "https://appassets.example/";
+                if (frameInfo.Source.StartsWith(virtualAppHostName))
+                {
+                    message = "Browser render process for app frame exited unexpectedly. Reload page?";
+                    caption = "App content frame unresponsive";
+                    AskForReload(message, caption);
+                    break;
+                }
+            }
+            break;
+        default:
+            // Show the process failure details. Apps can collect info for their logging purposes.
+            caption = "Child process failed";
+            StringBuilder messageBuilder = new StringBuilder();
+            messageBuilder.AppendLine($"Process kind: {e.ProcessFailedKind}");
+            messageBuilder.AppendLine($"Reason: {e.Reason}");
+            messageBuilder.AppendLine($"Exit code: {e.ExitCode}");
+            messageBuilder.AppendLine($"Process description: {e.ProcessDescription}");
+            MessageBox.Show(messageBuilder.ToString(), caption, MessageBoxButton.OK);
+            break;
+    }
+}
+
+```
+
+
+# Remarks
+* `ProcessFailedKind` in the event args will be `RENDER_PROCESS_EXITED` if the failed process is the main frame's renderer, even if there were subframes rendered by such process. All frames are gone when this happens and `ImpactedFrames` will be `null` as the attribute is intended for frame-only renderer failures only.
+
+* `Reason` in the event args is always `UNEXPECTED` when `ProcessFailedKind` is `BROWSER_PROCESS_EXITED`, and `UNRESPONSIVE` when `ProcessFailedKind` is `RENDER_PROCESS_UNRESPONSIVE`.
+
+* `ExitCode` is always `1` when `ProcessFailedKind` is `BROWSER_PROCESS_EXITED`, and `STILL_ACTIVE` (`259`) when `ProcessFailedKind` is `RENDER_PROCESS_UNRESPONSIVE`.
+
+
+# API Notes
+See [API Details](#api-details) section below for API reference.
+
+* Triple-slash (`///`) comments will appear in the public API documentation.
+* Double-slash (`//`) comments are notes for this review only and will not show in public documentation.
+
+
+# API Details
+## COM
+```cpp
+library WebView2
+{
+// ...
+
+/// Specifies the process failure type used in the
+/// `ICoreWebView2ProcessFailedEventHandler` interface. The values in this enum
+/// make reference to the process kinds in the Chromium architecture. For more
+/// information about what these processes are and what they do, see
+/// [Browser Architecture - Inside look at modern web browser](https://developers.google.com/web/updates/2018/09/inside-browser-part1)
+[v1_enum]
+typedef enum COREWEBVIEW2_PROCESS_FAILED_KIND {
+  // Existing stable values.
+  COREWEBVIEW2_PROCESS_FAILED_KIND_BROWSER_PROCESS_EXITED,
+
+  /// Indicates that the main frame's render process ended unexpectedly.  A new
```

Comment:
Weird that we only have comments for one of the three existing enum values. Is that just how it is in the IDL with no documentation for two of the enum values?

Reply:
No, we do have comments for the other values as well, but I wasn't including them as they're unchanged. I will copy them and add a note that they're unchanged.

Reply:
Added the comments.